mithril-common/src/digesters/immutable_file.rs

```diff
@@ -106,13 +106,14 @@ impl ImmutableFile {
         })
     }
 
-    /// ImmutableFile factory, TEST ONLY as it bypass the checks done by [ImmutableFile::new].
-    #[cfg(test)]
-    pub(crate) fn dummy(path: PathBuf, number: ImmutableFileNumber, filename: String) -> Self {
-        Self {
-            path,
-            number,
-            filename,
+    cfg_test_tools! {
+        /// ImmutableFile factory, TEST ONLY as it bypass the checks done by [ImmutableFile::new].
+        pub fn dummy(path: PathBuf, number: ImmutableFileNumber, filename: String) -> Self {
```

Comment:
(Optional) You could add a generic for the filename so it accept anything convertible to a string (to avoid the `.to_string()` when passing an str).
```suggestion
        pub fn dummy<T: Into<String>>(path: PathBuf, number: ImmutableFileNumber, filename: T) -> Self {
```

Reply:
Good idea, done :+1: